code/scripting/api/objs/parse_object.cpp

```diff
@@ -0,0 +1,124 @@
+            "The name of the object. If possible, don't set the name but set the display name instead.", "string",
+            "The name or empty string on error")
+{
+	parse_object_h* poh = nullptr;
+	const char* newName = nullptr;
+	if (!ade_get_args(L, "o|s", l_ParseObject.GetPtr(&poh), &newName))
+		return ade_set_error(L, "s", "");
+
+	if (poh == nullptr)
+		return ade_set_error(L, "s", "");
+
+	if (!poh->isValid())
+		return ade_set_error(L, "s", "");
+
+	if (ADE_SETTING_VAR) {
+		strcpy_s(poh->getObject()->name, newName);
+	}
+
+	return ade_set_args(L, "s", poh->getObject()->name);
+}
+
+ADE_VIRTVAR(
+    DisplayName, l_ParseObject, "string",
+    "The display name of the object. If the name should be shown to the user, use this since it can be translated.",
+    "string", "The display name or empty string on error")
+{
+	parse_object_h* poh = nullptr;
+	const char* newName = nullptr;
+	if (!ade_get_args(L, "o|s", l_ParseObject.GetPtr(&poh), &newName))
+		return ade_set_error(L, "s", "");
+
+	if (poh == nullptr)
+		return ade_set_error(L, "s", "");
+
+	if (!poh->isValid())
+		return ade_set_error(L, "s", "");
+
+	if (ADE_SETTING_VAR) {
+		poh->getObject()->display_name = newName;
+	}
+
+	return ade_set_args(L, "s", poh->getObject()->get_display_string());
+}
+
+ADE_VIRTVAR(Position, l_ParseObject, "vector", "The position at which the object will arrive.", "vector",
+            "The position of the object.")
+{
+	parse_object_h* poh = nullptr;
+	vec3d* newPos       = nullptr;
+	if (!ade_get_args(L, "o|o", l_ParseObject.GetPtr(&poh), l_Vector.GetPtr(&newPos)))
+		return ade_set_error(L, "o", l_Vector.Set(vmd_zero_vector));
+
+	if (poh == nullptr)
+		return ade_set_error(L, "o", l_Vector.Set(vmd_zero_vector));
+
+	if (!poh->isValid())
+		return ade_set_error(L, "o", l_Vector.Set(vmd_zero_vector));
+
+	if (ADE_SETTING_VAR) {
+		poh->getObject()->pos = *newPos;
+	}
+
+	return ade_set_args(L, "o", l_Vector.Set(poh->getObject()->pos));
+}
+
+ADE_VIRTVAR(Orientation, l_ParseObject, "matrix", "The orientation of the object.", "matrix", "The orientation")
+{
+	parse_object_h* poh = nullptr;
+	matrix_h* newMat    = nullptr;
+	if (!ade_get_args(L, "o|o", l_ParseObject.GetPtr(&poh), l_Matrix.GetPtr(&newMat)))
+		return ade_set_error(L, "o", l_Matrix.Set(matrix_h()));
+
+	if (poh == nullptr)
+		return ade_set_error(L, "o", l_Matrix.Set(matrix_h()));
+
+	if (!poh->isValid())
+		return ade_set_error(L, "o", l_Matrix.Set(matrix_h()));
+
+	if (ADE_SETTING_VAR) {
+		poh->getObject()->orient = *newMat->GetMatrix();
+	}
+
+	return ade_set_args(L, "o", l_Matrix.Set(matrix_h(&poh->getObject()->orient)));
+}
+
+ADE_VIRTVAR(ShipClass, l_ParseObject, "shipclass", "The ship class of the object.", "shipclass", "The ship class")
+{
+	parse_object_h* poh = nullptr;
+	int newClass        = -1;
+	if (!ade_get_args(L, "o|o", l_ParseObject.GetPtr(&poh), l_Shipclass.Get(&newClass)))
+		return ade_set_error(L, "o", l_Matrix.Set(matrix_h()));
```

Comment:
Copy-paste error?  I don't think matrices should be used here and on the next few lines.